public/js/tasks.js

```diff
@@ -0,0 +1,79 @@
+import {
+    inputEnabled,
+    setDiv,
+    message,
+    setToken,
+    token,
+    enableInput,
+  } from "./index.js";
+  import { showLoginRegister } from "./loginRegister.js";
+  import { showAddEdit } from "./addEdit.js";
+  
+  let tasksDiv = null;
+  let tasksTable = null;
+  let tasksTableHeader = null;
+  
+  export const handleTasks = () => {
+    tasksDiv = document.getElementById("tasks");
+    const logoff = document.getElementById("logoff");
+    const addTask = document.getElementById("add-task");
+    tasksTable = document.getElementById("tasks-table");
+    tasksTableHeader = document.getElementById("tasks-table-header");
+  
+    tasksDiv.addEventListener("click", (e) => {
+      if (inputEnabled && e.target.nodeName === "BUTTON") {
+        if (e.target === addTask) {
+          showAddEdit(null);
+        } else if (e.target === logoff) {
+            setToken(null);
+            message.textContent = "You have been logged off.";
+            tasksTable.replaceChildren([tasksTableHeader]);
+            showLoginRegister();
+        }
+      }
+    });
+  };
+  
+  export const showTasks = async () => {
+    try {
+        enableInput(false);
+        const response = await fetch("/api/tasks", {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                Authorization: `Bearer ${token}`,
+            },
+        });
+        const data = await response.json();
+        let children = [tasksTableHeader];
+
+        if(response.status === 200) {
+            if(data.count === 0) {
+                tasksTable.replaceChildren(...children);
+            } else {
+                for (let i=0; i < data.tasks.length; i++) {
```

Comment:
Whenever there are no tasks, an error occurs because your code can't access the `length` property. That's probably because tasks is `null` or `undefined`. Try setting an empty array as the default value when tasks could not be loaded.